### PR TITLE
implement findAndModify

### PR DESCRIPTION
**Pull request by DerNamenlose**
I need the `findAndModify` operation for a project, so I figured I'd try and contribute it to the driver. 

This also adds a few interfaces to lock down the interface in terms of types, that could be re-used in other existing interfaces like `findOne`, although I didn't change those yet.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -143,6 +143,108 @@ export interface UpdateOptions {
...
+}
+
+/**
+ * Options for the findAndModify operation
+ *
+ * @see https://docs.mongodb.com/manual/reference/method/db.collection.findAndModify/
+ */
+export interface FindAndModifyOptions<T> {
+  /**
+     * Control the order in which documents are found.
```
Comment (manyuanrong) [inline src/types.ts:198]: align those comment

**Comment by manyuanrong** [comment]
LGTM, a little formatting problem

File: src/types.ts (hunk 1 above)
Comment (DerNamenlose) [reply src/types.ts]: Fixed. Those must've slipped under the radar.

**Comment by manyuanrong** [comment]
I forgot something, can you add a test for the new method?

**Comment by DerNamenlose** [comment]
I'll give it a go. It'll probably take a few days, because I'll have to figure out the best way to to that first.

**Comment by DerNamenlose** [comment]
OK, that was quicker, than expected. I added two test cases, one for the update and one for the delete case.

**Comment by manyuanrong** [comment]
@DerNamenlose ci failed

**Comment by DerNamenlose** [comment]
Fixed. Funny enough: my `deno fmt --check` didn't find those initially. Must've called it wrong or something.

**Comment by manyuanrong** [comment]
> Fixed. Funny enough: my `deno fmt --check` didn't find those initially. Must've called it wrong or something.

Maybe it's a problem with the Deno version

**Pull request merged**